core/src/main/java/org/wso2/testgrid/infrastructure/InfrastructureCombinationsProvider.java

```diff
@@ -131,4 +136,44 @@ private void addSubpropertiesAsInfrastructureParameters(InfrastructureCombinatio
         }
     }
 
+    /**
+     * This class provides set of filtered infrastructure values.
+     * <p>
+     * The filtering is executed by reading exclude and include attributes in the @{@link TestgridYaml} file.
+     * If exclude attribute is set in @{@link TestgridYaml} and if those are exists in the current infrastructure
+     * value set, those will be eliminated.
+     * Further, if include attribute is set in @{@link TestgridYaml} and if those are exists in the current
+     * infrastructure value set, only those infrastructure will be consider when creating combinations.
+     *
+     * @param infrastructures entire infrastructure value set
+     * @param testgridYaml    object model of testgrid.yaml config file
+     * @return filtered infrastructure value set
+     */
+    private Set<InfrastructureValueSet> filterInfrastructures(Set<InfrastructureValueSet> infrastructures,
+            TestgridYaml testgridYaml) {
+
+        List<String> excludes = testgridYaml.getInfrastructureConfig().getExcludes();
+        List<String> includes = testgridYaml.getInfrastructureConfig().getIncludes();
+        Set<InfrastructureValueSet> selectedInfraValSet = ConcurrentHashMap.newKeySet();
+
+        if (excludes != null && !excludes.isEmpty()) {
+            infrastructures.forEach(infrastructureValueSet -> {
+                Set<InfrastructureParameter> selectedSet = infrastructureValueSet.getValues().stream()
+                        .filter(infrastructureParameter -> !excludes.contains(infrastructureParameter.getName()))
+                        .collect(Collectors.toSet());
+                selectedInfraValSet.add(new InfrastructureValueSet(infrastructureValueSet.getType(), selectedSet));
+            });
+            return selectedInfraValSet;
+        } else if (includes != null && !includes.isEmpty()) {
+            infrastructures.forEach(infrastructureValueSet -> {
+                Set<InfrastructureParameter> selectedSet = infrastructureValueSet.getValues().stream()
```

Comment:
shall we assign infrastructureValueSet.getValues() to a variable, It will improve the readability?

Reply:
IMO there is no any benefit here though we have introduced a separate variable for infrastructureValueSet.getValues() as we are not reusing this again within the code. :)